Bugfix for isdisjoint and performance improvement

mundya opened the pull request:
Two commits: the first fixes the implementation for `isdisjoint` which was inverted and adds a test for this (previously lacking); the second adds a faster implementation for `isdisjoint` which avoids creating another `intbitset` and returns as soon as it is clear that the two sets are not disjoint.

The test I've added doesn't follow the format of the other tests because I wasn't sure of the intent of the way the tests are set up.

mundya commented:
Grr Python 2.6, resubmitting changes to tests shortly.

kaplun commented:
@mundya thanks a lot for this contribution. I have indeed some comment on the testing, and I can give you more details on how the current testing works, so that you leverage the infrastructure (thus allowing you to test in many corner cases).

Coming soon with more info.

kaplun commented:
So the meaning of https://github.com/inveniosoftware/intbitset/blob/master/tests/test_intbitset.py#L65-L74 is:
```python
# (intbitset.__and__, set.__and__, int.__and__, False),
# intbitset operator to test, corresponding set operator (each intbitset operator should behave like the equivalent set operator, assuming a set of positive integers), corresponding operator to apply to the inifinite bit, finally whether the operator is changing the object in place (True) or is returning the result).
```
In your case, if I am not mistaken you could just add:
```python
intbitset.isdisjoint, set.isdisjoint, lambda x, y: not(x or y), False)
```
to the above lines and you could then be able to test `isdisjoint` on all the several input tests (also considering infinite sets).

kaplun commented:
Note, I have just merged #55 so you will need to rebase this PR (since it's currently failing on Python2.6 this is a bonus).

mundya commented:
Thanks @kaplun - I'll try to have a look at those changes later this afternoon.

pombredanne commented:
@mundya this would be mucho welcomed... and I reckon this has been waiting a long time but do you think you could rebase and address the comments?
Thanks!

mundya commented:
@pombredanne I think I changed jobs just after submitting the original PR, so it must have dropped off my radar. I've rebased onto master and made the changes to the tests suggested by @kaplun. Hopefully this should be good now!

mundya commented:
@pombredanne Would you mind having a look at this? Thanks!

pombredanne commented:
@mundya thanks ++

mundya commented:
Bump.

pombredanne commented:
@mundya At last ... sorry for the long lag! and thank you ++

mundya commented:
@pombredanne Thanks!
